lib/tenstorrent/bh_arc/eth.c

```diff
@@ -226,10 +234,31 @@ int LoadEthFw(uint32_t eth_inst, uint32_t ring, uint8_t *fw_image, uint32_t fw_s
  * @param fw_cfg_size Size of the FW config data
  * @return int 0 on success, -1 on failure
  */
-int LoadEthFwCfg(uint32_t eth_inst, uint32_t ring, uint32_t eth_enabled,
-	uint8_t *fw_cfg_image, uint32_t fw_cfg_size)
+int LoadEthFwCfg(uint32_t eth_inst, uint32_t ring, uint32_t eth_enabled, const uint8_t *tag)
 {
-	uint32_t *fw_cfg_32b = (uint32_t *)fw_cfg_image;
+	int rc;
+	tt_boot_fs_fd fd_data;
+
+	if (tt_boot_fs_find_fd_by_tag(flash, tag, &fd_data) != TT_BOOT_FS_OK) {
+		LOG_ERR("%s (%s) failed: %d", "tt_boot_fs_find_fd_by_tag", tag, -EIO);
+		return -EIO;
+	}
```

Comment:
It should be possible to re-use the return value from `tt_boot_fs_find_fd_by_tag()` on error.

```suggestion
	rc = tt_boot_fs_find_fd_by_tag(flash, tag, &fd_data);
	if (rc < 0) {
		LOG_ERR("%s (%s) failed: %d", "tt_boot_fs_find_fd_by_tag", tag, rc);
		return rc;
	}
```